### ISSUE TITLE
Unable to use Crow CLI

### ISSUE THREAD

**Issue by jmctune** [crow-translate/crow-translate#296]
**Bug description**
Running crow commands from the command prompt doesn't return any output. The help menu gets brought up, but trying to translate any text returns nothing.

**Steps to reproduce**
- Install the latest version of crow from the Releases page for Windows 
- In the installed Crow directory, run: `crow -t es hello`
- Nothing happens

**Expected behavior**
Expect to see the translated text output.

**Screenshots**
```
C:\Program Files\Crow Translate>crow -t es hello

C:\Program Files\Crow Translate>

C:\Program Files\Crow Translate>crow -c

C:\Program Files\Crow Translate>
```
**Environment**

-   OS: Windows 10 Pro Build 19041
-   Crow Translate version: 2.8.2

**Additional context**
UI works fine and I can translate text no problem.

**Comment by Shatur95** [comment]
Can confirm, happens only on Windows. This is because GUI applications on Windows detach from the console by default. Need to think about how to fix it.

**Comment by jmctune** [comment]
Just for the record, I tried every version as far back as 2.2.3 and this doesn't appear to have ever worked.

**Comment by Shatur95** [comment]
> Just for the record, I tried every version as far back as 2.2.3 and this doesn't appear to have ever worked.

Yes, sorry, just never used CLI on Windows. But I will try to fix it later.

**Comment by Shatur95** [comment]
I did some digging later and unfortunately this is not really a bug. Applications on Windows can't have GUI and CLI in a single executable. The OS needs to know before the program starts running which subsystem to use. You can check [this](https://devblogs.microsoft.com/oldnewthing/20090101-00/?p=19643) article.

But there are several workarounds:

1. We can build two separate utilities - GUI and CLI for Windows (just use `.com` extension for CLI-version).
2. Write a helper program that will have the `.com` extension and will launch the GUI and redirect the output. This is how it works in MPV and [Inkscape](http://bazaar.launchpad.net/~inkscape.dev/inkscape/trunk/view/head:/src/winconsole.cpp) ([wiki](https://wiki.inkscape.org/wiki/index.php?title=Frequently_asked_questions#I.27m_on_Windows.2C_and_command_line_parameters_don.27t_seem_to_work.21), but was changed after [b3fc73a](https://gitlab.com/inkscape/inkscape/-/commit/b3fc73a6008647b277e011c2c996be07ea8d5349)).
3. Try to attack to a parent CLI if available.  But the console continues accepting input on behalf of the parent process, and the parent process is not aware that it should wait for the child to finish running before using the console for other things.

It seems to me that the first method will be preferable. For example, Inkscape recently switched to it. I will need help implementing this.

**Comment by mcoret** [comment]
try:
crow -t es hello | more

**Comment by Shatur95** [comment]
>   crow -t es hello | more

Interesting workaround, works in `cmd` :)

**Comment by VolkMilit** [comment]
@Shatur, I talked to friend of mine, which is software-ingeener using Windows. He said, that there is [WinMain() entry point should be using for GUI](https://docs.microsoft.com/en-us/windows/win32/learnwin32/winmain--the-application-entry-point) and main() for everything else.

**Comment by Shatur** [comment]
Yep! This is why wee need a workaround with `.com` like in other projects.

**Comment by Suleman-Elahi** [comment]
Tried in terminal.... but encoding problem.

![image](https://user-images.githubusercontent.com/24384996/140015301-a1b1bda0-3acf-4dea-b205-6c78df7c8f5f.png)

**Comment by VolkMilit** [comment]
That's odd, because Qt return utf-8 (16 for windows) in all strings.

**Comment by CAESIUS-TIM** [comment]
It works in cmd but not in pwsh. 😕 

## cmd
![cmd](https://github.com/crow-translate/crow-translate/assets/60285058/0d7ac6d0-7f97-4886-9198-e93f0652940b)

## pwsh
![pwsh-encoding](https://github.com/crow-translate/crow-translate/assets/60285058/888141b8-86b6-4e9e-b560-3ea6b8ad0b5f)
![pwsh](https://github.com/crow-translate/crow-translate/assets/60285058/7196a2f4-ed04-445a-a186-e4e458cb9454)

**Comment by CAESIUS-TIM** [comment]
> Tried in terminal.... but encoding problem.
> 
> ![image](https://user-images.githubusercontent.com/24384996/140015301-a1b1bda0-3acf-4dea-b205-6c78df7c8f5f.png)

@Suleman-Elahi The problem is caused by `C:/WINDOWS/system32/more.com`.

> It works in cmd but not in pwsh. 😕

Since it works in cmd, and cmd's encoding id is:
shell: `cmd`
![cmd-chcp](https://github.com/crow-translate/crow-translate/assets/60285058/3b171484-fa8f-4b67-98fe-2c5933227de7)

It's a nice try to change pwsh's output encoding id into `936`.

shell: `pwsh`
![图片](https://github.com/crow-translate/crow-translate/assets/60285058/1c0a3451-926b-4a03-a1ab-7511133f6e6b)

**Comment by CAESIUS-TIM** [comment]
edit profile `notepad $PROFILE`
```ps1
function dict {
    param(
        [string] $text
    )
    
    [Console]::OutputEncoding = [System.Text.Encoding]::GetEncoding(936)
    crow -t en $text | more
    [Console]::OutputEncoding = [System.Text.Encoding]::UTF8
}
```
example
```pwsh
$ dict python
python
(?pī?TH?n)

[ 英语 -> 英语 ]

python

python - 示例：
        a large heavy-bodied nonvenomous snake occurring throughout the Old World tropics, killing prey by constriction and asphyxiation.

        a high-level general-purpose programming language.
```

**Comment by cu1tur4l** [comment]
Hi,
I know this is kinda not the place, but I honestly don't know where else to ask, well maybe on /mpv, but it also does not look like the right place... Anyway, is there anyone that could help me getting [this](https://gist.github.com/nainemom/0ac55c458f5a644959a069a16ccfb00b) to work? As I think the issue is the cli not working well on windows, I think it's **_KINDA_** related to crow git. But I understand if I'm rude to ask for help here, sorry then.
